### PR TITLE
Add Target to DND5E.targetTypes

### LINKED ISSUE #974: Add 'any' to DND5E.targetTypes
Originally in GitLab by @jeremyregnerus

D&D 5e uses the language **_target_** often in their stat blocks, which encompasses a creature, object or location as a single defined target type. This is different than targeting a creature or an object exclusively. This target type is not currently included in DND5E.targetTypes.

### LINKED ISSUE #1122: Add "willing creature" options for spell/feature/item targets
Originally in GitLab by @Fallayn

There are spells that can have either creatures or objects as their targets, while with Foundry 5E you can currently only select "Creature" or "Object" on the target option dropdown. I propose adding "Target" as a neutral option for this reason, or maybe "Creature/Object".

Some spells are only able to target willing creatures, I'd like to be able to make this clear by adding a "willing creature" option in addition to "creature".

### PR DESCRIPTION
**Pull request by tavernaDelBeerholder**
- Closes #974 
- Closes #1122 

A update version of the PR : https://github.com/foundryvtt/dnd5e/pull/1638

Adds a generic "Target" to the targetTypes - "Target" here refers to the group of creatures and objects, i.e. "you throw a fire bolt at one creature or object" or "this spell affects one target".

The actual fix is in the first commit - a single line that just adds the targetType.
The second commit updates all the SRD compendia packs with this option.
The third commit compiles the packs.

### CODE REVIEW

**Comment by p4535992** [comment]
@Fyorl Are there any problems with this PR ? I would add the new entries and possibly update the compendiums at a later time a little at a time to avoid problems with the big changes expected with 2.2.X, but at least in the meantime "make them official."

**Comment by Fyorl** [comment]
> @Fyorl Are there any problems with this PR ? I would add the new entries and possibly update the compendiums at a later time a little at a time to avoid problems with the big changes expected with 2.2.X, but at least in the meantime "make them official."

The PR looks small enough that we can consider it for 2.2

File: module/config.mjs [hunk 1]
```diff
@@ -876,7 +876,10 @@ DND5E.individualTargetTypes = {
   enemy: "DND5E.TargetEnemy",
   creature: "DND5E.TargetCreature",
   object: "DND5E.TargetObject",
-  space: "DND5E.TargetSpace"
+  space: "DND5E.TargetSpace",
+  single: "DND5E.TargetSingle",
```
[the remark this answers is not in the document]
Comment (p4535992) [reply module/config.mjs:880]: K I brushed up my memory a little bit if you ask me today "single" doesn't make much sense anymore I think it's ok to remove it .

**Comment by Fyorl** [comment]
We should add 'willing creature' here as well from #1122

**Comment by Fyorl** [comment]
Apologies, something evidently went wrong when I attempted to rebase this PR to add some additional changes, and GH has automatically closed it. I have re-created it here #2378. Please let me know if it is accurate.

**Pull request closed without merging**